libs-haskell/test-utils/BUILD.bazel

```diff
@@ -39,8 +39,6 @@ da_haskell_library(
     ],
     visibility = ["//visibility:public"],
     deps = [
-        "//compiler/daml-lf-ast",
-        "//compiler/damlc/stable-packages:stable-packages-lib",
```

Comment:
I remembered about this change yesterday while working on #18066 because a small test that innocently depended on `//libs-haskell/test-utils` took a surprising time to build. As I'd been working with `deps`, this looked like the immediate culprit, but ofc given `data = [ "//canton:community_app_deploy.jar" ]` some lines above, this will still be a bottle neck for tests that don't actually need Canton. Moving the Canton parts of `test-utils` into a separate lib and using that only when strictly needed would improve the situation a bit. To be clear, this is just a problem when developing locally, as CI will build everything at some point anyway.